File: CSharp/HtmlWalker/Exception/NoDocumentException.cs
namespace HtmlWalker.Exception
{
    /// <summary>
    ///   This exception is called when an atempt is made to walk HTML and the document
    ///   has been opened, <see cref="Walker.Open(WalkerPlatform.WalkerDocument)"/>.
    /// </summary>
    /// <author>Robert R Flores</author>
    public class NoDocumentException : HtmlWalkerException
    {
        public NoDocumentException() : base() { }

        public NoDocumentException(string msg) : base(msg) { }
    }
}

File: Java/src/htmlwalker/exception/NoDocumentException.java
/*
 * 
 */
package htmlwalker.exception;

/**
 * This exception is called when an atempt is made to walk HTML and the document
 * has not been set or made available.
 * 
 * Currently this document is intentially hidden, so it appears to the user as
 * its parent class <b>HtmlWalkerException</b>.
 * 
 * @author Robert Flores
 */
@SuppressWarnings("serial")
public class NoDocumentException extends HtmlWalkerException
{
	
	/**
	 * Instantiates a new no document exception.
	 */
	public NoDocumentException()
	{
		super(new IllegalStateException("IDocument not set"));
	}
	
	public NoDocumentException(String msg)
	{
		super(new IllegalStateException(msg));		
	}
}
